headless-Checkout/client/react/headless-checkout/src/pages/save-payment-method/SavePaymentMethod.jsx

```diff
@@ -1,7 +1,217 @@
+                didItPassMaxLengthValidation = value.length <= validations.max_length;
+            } else {
+                return true;
+            }
+            return didItPassMaxLengthValidation ? true : false;
+        }
+        return true;
+    }
+
+    function handleChange(e, field) {
+        let fieldAfterValueUpdate;
+        let inputEntry = e.target.value;
+        if(e.target.id === 'expiry' && !inputEntry.includes('/') && inputEntry.length === 2 && e.nativeEvent.data) {
+            fieldAfterValueUpdate = {...paymentDetails[e.target.id], value: `${e.target.value}/`};
+        } else {
+            fieldAfterValueUpdate = {...paymentDetails[e.target.id], value: e.target.value};
+        }
+
+        const fieldAfterValidation = {...fieldAfterValueUpdate, isValidated: validateField(fieldAfterValueUpdate.value, field.validations)}
+
+        setPaymentDetails({...paymentDetails, [e.target.id]: fieldAfterValidation});
+    }
+
+    function save() {
+        const paymentMethodOption = selectedPaymentMethod;
+        const formattedPaymentDetails = {
+            fields: []
+        };
+        let currentIndex = 0;
+        for(let key in paymentDetails){
+            formattedPaymentDetails.fields[currentIndex] = {
+                name: key,
+                value: paymentDetails[key].value
+            }
+            currentIndex++;
+        }
+        if (paymentMethodOption === 'card') {
+            formattedPaymentDetails.fields.push({name: 'save_card',value: true});
+        }
+        // create new instance of inai checkout
+        const inaiInstance = window.inai.create({
+            token: process.env.REACT_APP_CLIENT_USERNAME,
+            orderId: orderId,
+            countryCode: country
+        });
+
+        // invoke payment
+        inaiInstance.makePayment(paymentMethodOption, formattedPaymentDetails)
+        .then((data) => {
+            alert(JSON.stringify(data));
+            navigate('/headless-checkout-options');
+        })
+        .catch((err) => {
+            alert(JSON.stringify(err));
+        })
+    }
+
+    useEffect(() => {
+        getPaymentMethods();
+    }, []);
+
     return (
         <div>
-            save payment method page
+            <div className="font-weight-bold text-align-center mt-4">Payment Options</div>
+            {loading ? (
+                <div className="container w-20 text-align-center">loading...</div>
+            ) : null}
+            {error ? (
+                <div className="container w-20 text-align-center">{error}</div>
+            ) : null}
+            {shouldRenderPaymentMethodOptions ? (
+                <div className="container w-20 flex flex-column gap-10">
+                    {
+                        paymentMethodOptions.map((option, i) => (
+                            <div className="flex flex-column gap-10">
+                                <div key={`payment-method-${i}`} id={option.rail_code} className="btn btn-1" onClick={() => {
```

Comment:
Hey,  for css lets not give bootstrap or any classes here, just give one class name and write all styles in css files, if i am a user, and if i also get styles separately, i will be able to use them as per my need.

@inai-venthan If our motive is just demo, doesn't matter, but if we are exposing and letting them copy entire thing, jsx/js files should have minimal code, and css should be written entire separately.

Reply:
if not provided class as it is provided same css properties will be repeated in most of the css classes... in case merchant uses the same code and he is using some css library... the changes are overriden by our code... in case they want some other style... they can override.... @inai-venthan what should be the approach here?

Reply:
Can I take css changes  as asked in another pr separately as it involves changes at lot of places (in all the .jsx file of all the flows)?